src/main/kotlin/me/senseiwells/arucas/utils/error/ErrorUtils.kt

```diff
@@ -0,0 +1,56 @@
+package me.senseiwells.arucas.utils.error
+
+import me.senseiwells.arucas.core.Interpreter
+import me.senseiwells.arucas.utils.LocatableTrace
+import me.senseiwells.arucas.utils.Trace
+
+object ErrorUtils {
+    fun format(trace: Trace, interpreter: Interpreter?, message: String?): String? {
+         if (interpreter == null || trace.fileName != interpreter.name) {
+             // This interpreter does not belong to this trace
+             return null
+         }
+
+        return format(trace, interpreter.content, message)
+    }
+
+    fun format(trace: Trace, fileContent: String, message: String?): String? {
```

Comment:
You are always calling this method with a LocatableTrace anyways so might as well change the parameter, if you wanted to call this method and you didn't know I'd let the user do the type check instead of returning nullable.

Reply:
I made it take Trace because in the future this format could have other types of Traces. This also makes it easier so I do not have to check for the correct type every time this is called